### ISSUE TITLE
varying pointer to varying int incorrect result

### ISSUE THREAD

**Issue by caughtbyatoe** [ispc/ispc#1475]
This issue was first discussed on the ispc users group:
https://groups.google.com/forum/#!topic/ispc-users/FWwmNwfEMmY

When dealing with a varying pointer to a varying int the dereference operation is not respecting the active lane.  All the results appear to be dereferenced on lane 0.  If the int is wrapped in a struct, then the dereference happens as expected on the appropriate lanes.

As an example:
```
#ifdef USE_STRUCT
struct Point {
    int x;
};
#endif

export void
run()
{
#ifdef USE_STRUCT
    varying Point p[programCount];
    for (uniform int i = 0; i < programCount; ++i) {
        p[i].x = i * programCount + programIndex;
        print("% --> %, %\n", &p[i], p[i].x);
    }

    varying Point * uniform a = &p[0];
    print("%\n", a);
    varying Point * varying av = a;
    print("%\n", av);
    varying Point aDeref = *a;
    varying Point avDeref = *av;
    print("%\n", aDeref.x);
    print("%\n", avDeref.x);
#else
    varying int p[programCount];
    for (uniform int i = 0; i < programCount; ++i) {
        p[i] = i * programCount + programIndex;
        print("%\n", p[i]);
    }

    varying int * uniform a = &p[0];
    print("%\n", a);
    varying int * varying av = a;
    print("%\n", av);
    varying int aDeref = *a;
    varying int avDeref = *av;
    print("%\n", aDeref);
    print("%\n", avDeref);
#endif

}
```

When compiled with ispc-1.9.2 as:
ispc test.ispc --target=sse2-i32x4 -o test_ispc.o -h test_ispc.h

The following output is produced:
[0,1,2,3]
[4,5,6,7]
[8,9,10,11]
[12,13,14,15]
0x7ffd5f8fbac0
[0x7ffd5f8fbac0,0x7ffd5f8fbac0,0x7ffd5f8fbac0,0x7ffd5f8fbac0]
[0,1,2,3]
[0,0,0,0]

When compiled with ispc-1.9.2 as:
ispc test.ispc --target=sse2-i32x4 -o test_ispc.o -h test_ispc.h -DUSE_STRUCT

The following is produced:
0x7ffd8355c950 --> [0,1,2,3], 
0x7ffd8355c960 --> [4,5,6,7], 
0x7ffd8355c970 --> [8,9,10,11], 
0x7ffd8355c980 --> [12,13,14,15], 
0x7ffd8355c950
[0x7ffd8355c950,0x7ffd8355c950,0x7ffd8355c950,0x7ffd8355c950]
[0,1,2,3]
[0,1,2,3]

I haven't checked if the issue exists for other non-struct types.  But I suspect it does.  I also reproduced the issue with other target types (avx1-i32x8 and avx2-i32x8).  I included the sse build just because it is easier to read.

**Comment by caughtbyatoe** [comment]
I also tested with v1.11.0, same result

**Comment by dbabokin** [comment]
Fixed by #1475.
